yb-voyager/src/tgtdb/yugabytedb.go

```diff
@@ -243,8 +243,7 @@ func (yb *TargetYugabyteDB) InitConnPool() error {
 		SessionInitScript: getYBSessionInitScript(yb.tconf),
 	}
 	yb.connPool = NewConnectionPool(params)
-	redactedParams := params
-	redactedParams.ConnUriList = utils.GetRedactedURLs(redactedParams.ConnUriList)
+	redactedParams := *params
```

Comment:
We should not do this.
https://go.dev/play/p/NjUBR-pdyd7
This is typical example of shallow copy vs deep copy and can potentially lead to a bug in future again.

I think as a standard practice also, we should always do Deep copy unless there is a usecase for shallow or significant perf implications of deepcopy.


1. Either copy each field separately.
2. Make use of standard deepcopy libraries available (to address @makalaaneesh's concern)
https://github.com/mohae/deepcopy
https://github.com/barkimedes/go-deepcopy (this seems to be better maintained)

Reply:
Ah, good point! 
+1 to use deep copy by default. (unless there are pointers like mutexes/connections/etc. but then again if the struct has such fields, we wouldn't necessarily want to deep copy)

Reply:
Yeah, missed it thanks for bringing it up @sanyamsinghal .
I made this change but missed one more thing to do redaction of urls back for logging.
I am thinking to not blindly deep copy that to just for logging or for the case where we can also add new fields which don't need to be copied, we can add any new field to logged params as well if required. 
I can create the params with redaction for all current fields and log basically revert the commit

Reply:
Yeah you can copy each field manually also, just the overhead of maintaining the codepath for logging. But since the code is at one place so it will be fine.

btw do you see any issue with simply creating a deep copy and printing it? I believe this is a one time thing only?

Reply:
The issue with directly copying the params is that there will still be an overhead of redacting the URLs in ConnectionParams for logging, and this will still be present for any new fields we will add in the future to check if they need redaction. 
```
redactedParams := &ConnectionParams{}
deepcopy.Copy(redactedParams, params)
redactedParams.ConnUriList = utils.GetRedactedURLs(redactedParams.ConnUriList)
```
So, I think it is better to have the overhead of adding the new field to the logged parameters if required, as it is a sensitive part where we have connection info with passwords.

Reply:
We're probably overthinking this :) Let's go with whatever solves the problem easily.
IMO, deepcopy + redacting URL should be okay because:

1. We definitely want to log all connection params (so the full struct). 
2. Passwords are most likely the only thing we'd not want to log for security reasons. Everything else is okay (this is not callhome, it's just logs on the client machine).